lib/go_script/go.rb

```diff
@@ -90,10 +90,11 @@ def build_jekyll(extra_args = '')
   end
 
   def git_sync_and_deploy(commands, branch: nil)
-    exec_cmd 'git stash'
-    exec_cmd "git checkout -b #{branch}" unless branch.nil?
-    exec_cmd 'git pull'
-    exec_cmd 'bundle install' if File.exist? 'Gemfile'
+    exec_cmd 'git fetch origin #{branch}'
+    exec_cmd 'git clean -f'
+    exec_cmd "git reset --hard #{branch.nil? ? 'HEAD' : 'origin/' + branch}"
+    exec_cmd 'git submodule --sync --recursive'
+    exec_cmd 'git submodule update --init --recursive'
```

Comment:
Just to be clear, these two submodule commands are doing the following things:

1. Update the links to the remotes associated with the submodules themselves so that they are kept up-to-date and point to the correct repository(ies).
2.  Update each of the submodules themselves.

In both cases, recursively perform the operation for any other nested submodules.  Is my understanding correct?